connect/runtime/src/main/java/org/apache/kafka/connect/storage/OffsetStorageWriter.java

```diff
@@ -128,7 +127,7 @@ public synchronized boolean beginFlush() {
      *
      * @return a Future, or null if there are no offsets to commitOffsets
      */
-    public Future<Void> doFlush(final Callback<Void> callback) {
+    public synchronized Future<Void> doFlush(final Callback<Void> callback) {
```

Comment:
I think we might want to avoid holding the lock when invoking `backingStore.set` if it's not needed. The problem is that the lock is also needed in the callback. This can lead to deadlock if the backing store implementation uses its own lock. For example, if this lock is A and the backing store uses lock B, then this method will result in A-B  lock order. But the asynchronous callback invocation might result in B-A. I _think_ this is not a problem for any of the backing store implementations we currently have, but we may as well be a little defensive about future implementations.

Reply:
Thanks, I excluded `backingStore.set` from sync block.